src/DarkWeb/DarkWeb.tsx

```diff
@@ -86,13 +85,17 @@ export function buyDarkwebItem(itemName: string): void {
 
 export function buyAllDarkwebItems(): void {
   const itemsToBuy: DarkWebItem[] = [];
-  let cost = 0;
 
   for (const key of Object.keys(DarkWebItems) as (keyof typeof DarkWebItems)[]) {
     const item = DarkWebItems[key];
     if (!Player.hasProgram(item.program)) {
       itemsToBuy.push(item);
-      cost += item.price;
+      if (item.price > Player.money) {
+        Terminal.error("Not enough money to purchase remaining programs");
```

Comment:
```suggestion
        Terminal.error("Not enough money to purchase remaining programs, " + formatMoney(cost) + "required for " + item.program);
```
Lint might want to re-wrap that.

Reply:
how about telling them how much they actually need and not just the total cost of the program?

![image](https://github.com/bitburner-official/bitburner-src/assets/147098375/c43f7035-4886-4642-bd9f-7d5b955837bb)

Reply:
That works too, I just wanted there to be feedback on what you couldn't buy.

Reply:
cool, code submitted